internal/session/query.go

```diff
@@ -424,64 +424,260 @@ where session_id = ?
 	and credential_sha256 = ?;
 `
 	listSessionsTemplate = `
-with session_ids as (
-    select public_id
+with sessions as (
+    select *
       from session
            -- search condition for applying permissions is constructed
      where %s
   order by create_time desc, public_id asc
      limit %d
+),
+session_host_set_hosts as (
+    select session_id,
+           host_id,
+           host_set_id
+      from session_host_set_host
+     where session_id in (select public_id from sessions)
+),
+session_worker_protocols as (
+    select session_id,
+           worker_id
+      from session_worker_protocol
+     where session_id in (select public_id from sessions)
+),
+session_states as (
+    select session_id,
+           state,
+           previous_end_time,
+           start_time,
+           end_time
+      from session_state
+     where session_id in (select public_id from sessions)
 )
-   select *
-     from session_list
-    where session_list.public_id in (select * from session_ids)
+   select s.public_id,
+          s.user_id,
+          shsh.host_id,
+          s.target_id,
+          shsh.host_set_id,
+          s.auth_token_id,
+          s.project_id,
+          s.certificate,
+          s.certificate_private_key,
+          s.expiration_time,
+          s.connection_limit,
+          s.tofu_token,
+          s.key_id,
+          s.termination_reason,
+          s.version,
+          s.create_time,
+          s.update_time,
+          s.endpoint,
+          s.worker_filter,
+          s.egress_worker_filter,
+          s.ingress_worker_filter,
+          swp.worker_id,
```

Comment:
As @tmessi mentioned, it seems we could drop this from the query as it's not used in constructing the session: https://github.com/hashicorp/boundary/blob/da5dd383ad08d3977bdf94a40158526238b0f514/internal/session/repository.go#L143-L164. That means we could drop the whole `session_worker_protocols` CTE. Could you do a pass over the other columns here and see if there's anything else we can drop?